[languages] Caching, Logging and Typography in Flutter  resolves issue #4523

somtobe opened the pull request:
### Article checklist

## Attention
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

#### Pre-submission checks
> Please ensure that you have only one open topic suggestion (issue) + in-review article (linked pull request) at a time. This will ensure that we complete the review process in a timely manner from inception to publishing

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [ ] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
> Remove the backticks and add the issue number below to link and close your Topic Suggestion (issue) when your article has been published (PR has been merged). See this [video](https://youtu.be/TKJ4RdhyB5Y?t=278) for more details.

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

WanjaMIKE commented:
Plagiarism check done.

BrianaNzivu commented on content/articles/caching-logging-typography-in-flutter/index.md:
````diff
@@ -0,0 +1,750 @@
+}
+```
+
+Then output basin info and fine message using the methods provided by logging package. 
````
This statement is unclear

BrianaNzivu reviewed (changes requested):
Hey @somtobe  👋 .
My name is Briana and I will be reviewing your article.
First of all, good job on the article. It is a very interesting read and very minimal corrections to be made.
Just a few changes I need you to make that I have mentioned as we continue with the tech review.
1.  Please structure your article according to this [sample](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md). I need you to clearly highlight your steps.
2.  Refer to the single comment I have made.

**Note: Please make sure to make changes on the updated index.md**

In case of any question or help needed please do not hesitate to ping me here :smile:.

Regards.

Eze4Manuel commented:
Hello Briana Nzivu,
I have made the updates you requested. The lapse code is the same method as the previous code. So to avoid duplication and enable easy understanding for users, I have added comments to the code, informing the reader of what is happening there. I also change the arrangement to fit the format for the link you sent

somtobe commented:
Hello Briana Nzivu,
I have updated the index.md file. For the section for the comment you made I have add some comments to the code that enlightens the reader on what to do in that section. For the file update, I have updated the file such that they follow the sequence in the article link you sent.

BrianaNzivu commented:
Tech review done 👍

BrianaNzivu commented on content/articles/caching-logging-typography-in-flutter/index.md:
```diff
@@ -0,0 +1,759 @@
+
+Based on the information gotten from the constructor, we store that in a data field, making it accessible to the class.
+
+#### Step 3
```
Explain the title properly i.e Step three: Adding this and this in our application

somtobe commented:
I have made the updates and added explanation to the steps

BrianaNzivu reviewed (approved):
Hey @somtobe  👋 .

Very good job on the article. It is a very interesting read 👏 .

Just a few things I'd like to point out:
- I suggest that you check out the [structure](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) for the various articles accepted by Section.
-  It is always advisable to submit an article that is well polished. By this, I mean having an article that does not have too many grammatical corrections and functional code. I suggest that you check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) page which has links to various resourceful tools for writing. 
- Lastly please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

hectorkambow commented:
Jumping on this one!
